docs/architecture/adr-111-nil-mempool.md

```diff
@@ -0,0 +1,239 @@
+# ADR 111: `nil` Mempool
+
+## Changelog
+
+- 2023-11-07: First version (@sergio-mena)
+
+## Status
+
+Accepted | Rejected | Deprecated | Superseded by
+
+## Context
+
+### In the Beginning...
```

Comment:
I'd recommend putting all this content under the Detailed Design section, and have a very brief overview/summary here under Context. That could be another good opportunity to emphasize how the impact here is to fully enable app-side mempools.

Reply:
I'll add a short section on top of this one with a brief overview, as you suggest. But I'll leave this here, as it is context (not design)